### PR TITLE
 Restrain DynamicAtlas in case of initial polygon

### PR DESCRIPTION
**Pull request by matthieun**
In DynamicAtlas, when the initial shards are supplied with a Polygon or MultiPolygon, and the expansion policy is withExtendIndefinitely=false, then the expansion needs not consider the full initial shards for intersection, but the initial polygon.

This PR fixes that issue. Also added one test.

### CODE REVIEW

File: src/test/java/org/openstreetmap/atlas/geography/atlas/dynamic/DynamicAtlasRestrainedExpansionWithPolygonTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        this.store.put(new SlippyTile(6, 34, 6),
+                this.rule.getAtlas().subAtlas(new SlippyTile(6, 34, 6).bounds()).get());
+        this.store.put(new SlippyTile(4, 34, 6),
+                this.rule.getAtlas().subAtlas(new SlippyTile(4, 34, 6).bounds()).get());
+        this.dynamicAtlas = new DynamicAtlas(policy);
+        this.dynamicAtlas.preemptiveLoad();
+    }
+
+    @Test
+    public void testNotMovingTooFast()
```
Comment (MikeGost) [inline src/test/java/org/openstreetmap/atlas/geography/atlas/dynamic/DynamicAtlasRestrainedExpansionWithPolygonTest.java]: Nit: I'm probably not familiar with the terminology, the test name is a little hard to follow. It sounds like we're testing DynamicAtlas expansion based off the features that extend out of the starting (Multi)Polygon instead of features that extend out of all the shards that intersect with the starting Polygon. Maybe rename it to "testFeatureExpansionWithPolygon" or "testRestrainedExpansion"?

Comment (matthieun) [reply src/test/java/org/openstreetmap/atlas/geography/atlas/dynamic/DynamicAtlasRestrainedExpansionWithPolygonTest.java]: Oh! That is a copy paste error! Thanks!
